support/certs/tls.go

```diff
@@ -166,37 +157,25 @@ func signedCertificate(
 		Version:               3,
 		BasicConstraintsValid: true,
 	}
-	pub := caCert.PublicKey.(*rsa.PublicKey)
-	certTmpl.SubjectKeyId, err = generateSubjectKeyID(pub)
+
+	certTmpl.SubjectKeyId, err = rsaPubKeySHA1Hash(&key.PublicKey)
```

Comment:
@stlaz, we only use RSA?

Reply:
the private keys on the input of both the functions are RSA so it's safe to assume this